### PR TITLE
OAuth

### PR DESCRIPTION
**Pull request by samolubode**
This PR addressed #13, and I:
- Added Google and Github OAuth2
- Modified the User schema: 
 - - Added 'socialId' field to  store the profile id of users using social login(OAuth).
 - -  Changed 'allowNull' property in username, email and password to true, to save social credentials that may not contain some of these fields.

### CODE REVIEW

File: controllers/authController.js [hunk 1]
```diff
@@ -11,84 +11,112 @@ const User = db.users;
...
-  // console.log(req.headers)
-  // logger.info(req.body)
-  const { username, email, password, displayName } = req.body;
+    // console.log(req.headers)
+    // logger.info(req.body)
 
-  // // validate input
-  if (!(username && email && password && displayName))
-    throw new AppError("All fields are required", 400);
+    const payload = {};
```
Comment (KorkuTegbe) [inline controllers/authController.js:38]: please what's this 'payload' object?

**Review by KorkuTegbe — COMMENTED** [review]
`const payload = {}`
what is it?

File: controllers/authController.js (hunk 1 above)
Comment (samolubode) [reply controllers/authController.js:38]: I left it there by mistake. Will rectify that.

Comment (KorkuTegbe) [reply controllers/authController.js:38]: alright, sure

**Comment by ademesojosiah** [comment]
Love the code structuring still going through it sha 
👍

**Pull request merged**